src/senaite/core/exportimport/instruments/importer.py

```diff
@@ -0,0 +1,1049 @@
+        return api.get_object(results[0])
+
+    def get_attachment_type_by_title(self, title):
+        """Get an attachment type by title
+
+        :param title: Attachment type title
+        :returns: Attachment object or None
+        """
+        query = {
+            "portal_type": "AttachmentType",
+            "title": title,
+            "is_active": True,
+        }
+        results = self.setup_catalog(query)
+        if not len(results) > 0:
+            return None
+        return api.get_object(results[0])
+
+    def process(self):
+        parsed_results = self.parse_results()
+
+        # no parsed results, return
+        if not parsed_results:
+            return False
+
+        # Log allowed sample and analyses states
+        self.log(_("Allowed sample states: ${allowed_states}", mapping={
+            "allowed_states": ", ".join(self.allowed_sample_states_msg)
+        }))
+        self.log(_("Allowed analysis states: ${allowed_states}", mapping={
+            "allowed_states": ", ".join(self.allowed_analysis_states_msg)
+        }))
+        if not any([self.override_non_empty, self.override_with_empty]):
+            self.log(_("Don't override analysis results"))
+        if self.override_non_empty:
+            self.log(_("Override non-empty analysis results"))
+        if self.override_with_empty:
+            self.log(_("Override non-empty analysis results, also with empty"))
+
+        # Attachments will be created in any worksheet that contains
+        # analyses that are updated by this import
+        attachments = {}
+        infile = self.parser.getInputFile()
+
+        ancount = 0
+        updated_analyses = []
+        importedinsts = {}
+        importedars = {}
+
+        for sid, results in parsed_results.items():
+            # get the sample
+            sample = self.get_sample_by_id(sid)
+            refsample = None
+
+            # fetch all analyses for the given sample ID
+            analyses = self.get_analyses_for(sid)
+
+            # No registered analyses found, but maybe we need to
+            # create them first if we have an instrument
+            if len(analyses) == 0 and not self.instrument:
+                self.warn(_("Instrument not found"))
+                self.warn(_("No Sample with '${allowed_ar_states}' states"
+                            "found, and no QC analyses found for ${sid}",
+                            mapping={
+                                "allowed_ar_states": ", ".join(
+                                    self.allowed_sample_states_msg),
+                                "sid": sid}))
+                continue
+
+            # we have an instrument
+            elif len(analyses) == 0 and self.instrument:
+                # Create a new ReferenceAnalysis and link it to the Instrument.
+                refsample = self.get_reference_sample_by_id(sid)
+                if not refsample:
+                    self.warn(_("No Reference Sample found for ${sid}",
```

Comment:
Replace by "No Sample found for ${sid}" (remove "Reference"), cause this condition will be true also when no *routine* analyses (with or without sample) are found, that will probably happen more often

Reply:
Updated the message in 9b1f2c04d